ts/components/session/SessionChannelSettings.tsx

```diff
@@ -50,15 +50,19 @@ export class SessionChannelSettings extends React.Component<Props, any> {
   }
 
   public componentDidUpdate() {
-    this.getMediaGalleryProps()
-      .then(({ documents, media, onItemClick }) => {
-        this.setState({
-          documents,
-          media,
-          onItemClick,
-        });
-      })
-      .ignore();
+    const mediaScanInterval = 1000;
+
+    setTimeout(() => {
+      this.getMediaGalleryProps()
```

Comment:
A few things:

* `setTimeout` will be triggered only once after the timeout expires, why call the delay an `interval` then?
* if you do want to use an interval to periodically refresh the documents and gallery, be sure to call the `setInterval` in the `componentDidMount` and not the `componentDidUpdate` method. Otherwise, you will have more and more intervals running in parrallel as the component gets updated.

Reply:
Component was refreshing over 30x per second, because you changed state within the componentWillUpdate method; causing an infinite loop. The setTimeout slows this down.